var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -442,7 +442,7 @@ def configure_options(self):
 
         # Activates library support if needed (i.e. direct dependency)
         jpeg_sp = spec['jpeg'].name if 'jpeg' in spec else 'jpeg'
-        scalapack_sp = spec['scalapack'].name if 'scalapack' in spec else 'scalapack'
+        scalapack_sp = 'scalapack'
```

Comment:
I don't remember why `scalapack_sp` was needed before [and which use case was broke without it] - but if its not needed anymore - the following change would be better:

```diff
diff --git a/var/spack/repos/builtin/packages/petsc/package.py b/var/spack/repos/builtin/packages/petsc/package.py
index 547d050375..853650c4ec 100644
--- a/var/spack/repos/builtin/packages/petsc/package.py
+++ b/var/spack/repos/builtin/packages/petsc/package.py
@@ -442,7 +442,6 @@ def configure_options(self):
 
         # Activates library support if needed (i.e. direct dependency)
         jpeg_sp = spec['jpeg'].name if 'jpeg' in spec else 'jpeg'
-        scalapack_sp = spec['scalapack'].name if 'scalapack' in spec else 'scalapack'
 
         # to be used in the list of libraries below
         if '+fortran' in spec:
@@ -489,7 +488,7 @@ def configure_options(self):
                 ('libyaml', 'yaml', True, True),
                 'hwloc',
                 (jpeg_sp, 'libjpeg', True, True),
-                (scalapack_sp, 'scalapack', False, True),
+                ('scalapack', 'scalapack', False, True),
                 'strumpack',
                 'mmg',
                 'parmmg',

Reply:
@balay Ok. I'll modify it.